### ISSUE TITLE
when a backup storage location is deleted, remove associated backups & restic repos from the cluster

### ISSUE THREAD

**Issue by grozan** [vmware-tanzu/velero#2082]
**What steps did you take and what happened:**
cluster migration scenario, using velero v1.2.0.
On my new-cluster, I add a read-only second storage-location (containing the backups of the old-cluster). I restore my backups from the old-cluster to the new-cluster. When done, I delete the second storage-location on my new-cluster.

problem: `velero backup get` still shows me the backups I have on the storage-location I have now removed. The restic repositories are still known as well
`kubectl -n velero get backups.velero.io` and `resticrepositories.velero.io` still show stuff from the storage-location that does not show up anymore in the output of `kubectl -n velero get backupstoragelocations.velero.io`

**What did you expect to happen:**

after I run `kubectl -n velero delete backupstoragelocations.velero.io old-cluster`, I expect to not see anything related to `old-cluster`.

Also, the command-line tool should support a more intuitive `velero backup-location delete old-cluster` call IMO

**Comment by drazul** [comment]
Also, as related topic, we can't delete a backup or snapshot location using the command like tool.

`velero backup-location -h` only shows `create` and `get` commands.

```bash
$velero version
Client:
        Version: v1.4.2
        Git commit: -
Server:
        Version: v1.4.0
```

**Comment by HaveFun83** [comment]
same here
```
velero version   
Client:
	Version: v1.4.0
	Git commit: 5963650c9d64643daaf510ef93ac4a36b6483392
Server:
	Version: v1.4.2
```

when we delete the old backuplocation the velero pod log errors like:
```
time="2020-10-23T14:14:18Z" level=error msg="Error checking repository for stale locks" controller=restic-repository error="error getting backup storage location: backupstoragelocation.velero.io \"main1\" not found" error.file="/github.com/vmware-tanzu/velero/pkg/restic/common.go:235" error.function=github.com/vmware-tanzu/velero/pkg/restic.GetCACert logSource="pkg/controller/restic_repository_controller.go:141" name=prod-main1-75cp7 namespace=velero
```

**Comment by carlisia** [comment]
@drazul and @HaveFun83: we have recently added a sub command to delete a BSL (https://github.com/vmware-tanzu/velero/pull/3073). It'll be available with v1.6.

However, a BSL is only a representation of the storage, and deleting it will never delete the backups uploaded to that storage. This is intentional. 

@drazul: If you need to delete backups from a deleted BSL, create a new BSL pointing to that same storage.

As for the original request, which is to have a way to force delete backups associated with a BSL when that BSL is deleted, this is still in the backlog.

**Comment by nrb** [comment]
Let's tackle this alongside #2697, they're very similar
